pkg/boot/stboot/bootball.go

```diff
@@ -0,0 +1,574 @@
+	if err != nil {
+		return ball, fmt.Errorf("BootBall: getting configuration faild: %v", err)
+	}
+
+	dir, err := makeConfigDir(cfg, filepath.Dir(configFile))
+	if err != nil {
+		return ball, fmt.Errorf("BootBall: creating standard configuration directory faild: %v", err)
+	}
+
+	ball.Archive = archive
+	ball.dir = dir
+	ball.config = cfg
+
+	err = ball.init()
+	if err != nil {
+		return ball, err
+	}
+
+	return ball, nil
+}
+
+func (ball *BootBall) init() (err error) {
+	cert, err := getRootCert(filepath.Join(ball.dir, signaturesDirName, rootCertName))
+	if err != nil {
+		return fmt.Errorf("BootBall: getting configuration faild: %v", err)
+	}
+
+	bootFiles, err := getBootFiles(ball.config, ball.dir)
+	if err != nil {
+		return fmt.Errorf("BootBall: getting boot files faild: %v", err)
+	}
+
+	ball.rootCert = cert
+	ball.numBootConfigs = len(ball.config.BootConfigs)
+	ball.bootFiles = bootFiles
+	ball.Signer = Sha512PssSigner{}
+
+	err = ball.getSignatures()
+	if err != nil {
+		return fmt.Errorf("BootBall: getting signatures: %v", err)
+	}
+
+	var x int = 0
+	for _, sigPool := range ball.signatures {
+		if x == 0 {
+			x = len(sigPool)
+			continue
+		}
+		if len(sigPool) != x {
+			return errors.New("BootBall: invalid map of signatures")
+		}
+	}
+	ball.NumSignatures = x
+	return
+}
+
+// Clean removes the underlaying temporary directory.
+func (ball *BootBall) Clean() (err error) {
+	err = os.RemoveAll(ball.dir)
+	if err != nil {
+		return
+	}
+	ball.dir = ""
+	return
+}
+
+// Pack archives the all contents of the underlaying temporary
+// directory using zip.
+func (ball *BootBall) Pack() (err error) {
+	if ball.Archive == "" || ball.dir == "" {
+		return errors.New("BootBall.Pacstandak: booball.archive and bootball.dir must be set")
+	}
+	return toZip(ball.dir, ball.Archive)
+}
+
+// Dir returns the temporary directory associated with BootBall.
+func (ball *BootBall) Dir() (dir string) {
+	return ball.dir
+}
+
+// GetBootConfigByIndex returns the Bootconfig at index from the BootBall's configs arrey.
+func (ball *BootBall) GetBootConfigByIndex(index int) (bc *bootconfig.BootConfig, err error) {
+	bc, err = ball.config.getBootConfig(index)
+	if err != nil {
+		return
+	}
+	bc.SetFilePathsPrefix(ball.dir)
+	return
+}
+
+// Hash calculates hashes of all boot configurations in BootBall using the
+// BootBall.Signer's hash function
+func (ball *BootBall) Hash() (err error) {
+	ball.hashes = make(map[string][]byte)
+	for i, files := range ball.bootFiles {
+		hash, herr := ball.Signer.hash(files...)
+		if herr != nil {
+			return herr
+		}
+		ball.hashes[i] = hash
+	}
+	return
+}
+
+// Sign signes the hashes of all boot configurations in BootBall using the
+// BootBall.Signer's hash function with the provided privKeyFile. The signature
+// is stored along with the provided certFile inside the BootBall.
+func (ball *BootBall) Sign(privKeyFile, certFile string) (err error) {
+	err = validateFiles("", privKeyFile, certFile)
```

Comment:
I thought it would be an elegant way to avoid checks at other places in the code. But you're right. Whenever a file is opened, existence should be checked. Fixed it.